src/react/Router.tsx

```diff
@@ -151,60 +120,152 @@ function InterfaceEdit(props) {
   );
 }
 
-function RegisterDevice(props) {
+function RegisterDevice(props: PageProps): React.ReactElement {
   const searchQuery = new URLSearchParams(useLocation().search);
   const deviceId = searchQuery.get('deviceId') || '';
 
   return <RegisterDevicePage deviceId={deviceId} {...props} />;
 }
 
-function GroupDevicesSubPath(props) {
+function GroupDevicesSubPath(props: PageProps): React.ReactElement {
   const { groupName } = useParams();
   const decodedGroupName = decodeURIComponent(groupName);
 
   return <GroupDevicesPage groupName={decodedGroupName} {...props} />;
 }
 
-function FlowDetails(props) {
+function FlowDetails(props: PageProps): React.ReactElement {
   const { flowName } = useParams();
 
   return <FlowDetailsPage flowName={flowName} {...props} />;
 }
 
-function FlowConfiguration(props) {
+function FlowConfiguration(props: PageProps): React.ReactElement {
   const { pipelineId } = useParams();
 
   return <FlowConfigurationPage pipelineId={pipelineId} {...props} />;
 }
 
-function PipelineSubPath(props) {
+function PipelineSubPath(props: PageProps): React.ReactElement {
   const { pipelineId } = useParams();
 
   return <PipelineSourcePage pipelineId={pipelineId} {...props} />;
 }
 
-function DeviceStatusSubPath(props) {
+function DeviceStatusSubPath(props: PageProps): React.ReactElement {
   const { deviceId } = useParams();
 
   return <DeviceStatusPage deviceId={deviceId} {...props} />;
 }
 
-function BlockSubPath(props) {
+function BlockSubPath(props: PageProps): React.ReactElement {
   const { blockId } = useParams();
 
   return <BlockSourcePage blockId={blockId} {...props} />;
 }
 
-function DeviceDataSubPath(props) {
+function DeviceDataSubPath(props: PageProps): React.ReactElement {
   const { deviceId, interfaceName } = useParams();
 
   return <DeviceInterfaceValues deviceId={deviceId} interfaceName={interfaceName} {...props} />;
 }
 
-function NoMatch({ fallback }) {
+interface NoMatchProps {
+  fallback: (requesteURL: string) => void;
+}
+
+function NoMatch({ fallback }: NoMatchProps): React.ReactElement {
   const pageLocation = useLocation();
   const relativeUrl = [pageLocation.pathname, pageLocation.search, pageLocation.hash].join('');
   fallback(relativeUrl);
 
   return <p>Redirecting...</p>;
 }
+
+interface Props {
+  reactHistory: History;
+  astarteClient: AstarteClient;
+  sessionManager: SessionManager;
+  config: any;
+  fallback: (requesteURL: string) => void;
+}
+
+interface HistoryState {
+  action: HistoryAction;
+  location: HistoryLocation;
+}
+
+export default ({
+  reactHistory: history,
+  astarteClient,
+  sessionManager,
+  config,
+  fallback,
+}: Props): React.ReactElement => {
+  const [historyState, dispatchHistoryUpdate] = useReducer(
+    (_: unknown, action: HistoryState) => action,
```

Comment:
i think you can skil the first argument entirelly `(, action: HistoryState) => `
`_: unknown` is more explicit but doesn't provide any usefull information.
No strong opinions here, maybe we should opt for one style and document it for future similar cases

Reply:
Unfortunately if you type `(, action: HistoryState) =>` Typescript starts screaming. However since the two arguments are exactly the same type, I just went with `(prevState: HistoryState, state: HistoryState) =>` so it's clear what is going on.